index.html

```diff
@@ -6511,32 +6497,25 @@ <h3>Executing Script</h3>
   in <var>value</var> has been destroyed,
   return <a>error</a> with <a>error code</a> <a>stale element reference</a>.
 
-  <p>Otherwise return <a>success</a> with data set
-   to <a><code>WindowProxy</code> reference object</a>
-   for <var>value</var>.
+  <p>Otherwise return <a>success</a> with data <a><code>WindowProxy</code>
+   reference object</a> for <var>value</var>.
 
  <dt>has an <a>own property</a> named "<code>toJSON</code>" that is
   a <a>Function</a>
- <dd>Return <a>success</a> with the value returned by
+ <dd><p>Let <var>result</var> be the return value of
   <a>Function.[[\Call]]</a>(<code>toJSON</code>) with <var>value</var>
-  as the this value.
+  as the this value. If doing so causes the "<code>toJSON</code>"" script to throw
+  an error, return <a>error</a> with <a>error code</a> <a>javascript error</a>.
+
+  <p>Otherwise return <a>success</a> with data <var>result</var>.
 
- <dt>Otherwise</dt>
+ <dt>Otherwise
  <dd>
   <ol>
-   <li><p>If <var>value</var> is in <var>seen</var>,
-    return <a>error</a> with <a>error code</a> <a>javascript error</a>.
-
-   <li><p>Append <var>value</var> to <var>seen</var>.
-
-   <li><p>Let <var>result</var> be the value of running the
+   <li><p>Return <a>success</a> with data return value of running the
```

Comment:
Let's set the return value of the algorithm to a variable and return data with that.